torsiondrive/td_api.py

```diff
@@ -21,15 +21,16 @@ class DihedralScanRepeater(DihedralScanner):
     the requirements of torsiondrive-API"""
     def repeat_scan_process(self):
         """ Mimicing DihedralScanner.master function, but stops when new jobs needs to run """
-        self.push_initial_opt_tasks()
-        if len(self.opt_queue) == 0:
-            print("No tasks in opt_queue! Exiting..")
-            return
         # make sure we're in the rootpath
         os.chdir(self.rootpath)
         self.refined_grid_ids = set()
         self.next_jobs = defaultdict(list)
         self.current_finished_job_results = PriorityQueue()
+        # push the initial tasks
+        self.push_initial_opt_tasks()
+        if len(self.opt_queue) == 0:
+            print("No tasks in opt_queue! Exiting..")
```

Comment:
If you ever get the chance, it would be good to move all of the print statements to logging statements so that we can capture it. See the JSON runner in GeomeTRIC for an example.